vortex-array/src/array/byte_bool/compute/mod.rs

```diff
@@ -0,0 +1,242 @@
+    },
+    encoding::ArrayEncodingRef,
+    stats::StatsSet,
+    validity::ArrayValidity,
+    ArrayDType, ArrayData, ArrayTrait, IntoArray,
+};
+
+impl ArrayCompute for ByteBoolArray {
+    fn as_arrow(&self) -> Option<&dyn crate::compute::as_arrow::AsArrowArray> {
+        Some(self)
+    }
+
+    fn compare(&self) -> Option<&dyn crate::compute::compare::CompareFn> {
+        Some(self)
+    }
+
+    fn fill_forward(&self) -> Option<&dyn crate::compute::fill::FillForwardFn> {
+        None
+    }
+
+    fn scalar_at(&self) -> Option<&dyn ScalarAtFn> {
+        Some(self)
+    }
+
+    fn slice(&self) -> Option<&dyn crate::compute::slice::SliceFn> {
+        Some(self)
+    }
+
+    fn take(&self) -> Option<&dyn crate::compute::take::TakeFn> {
+        Some(self)
+    }
+}
+
+impl ScalarAtFn for ByteBoolArray {
+    fn scalar_at(&self, index: usize) -> VortexResult<Scalar> {
+        if index >= self.len() {
+            vortex_bail!(OutOfBounds: index, 0, self.len());
+        }
+
+        let scalar = match self.is_valid(index).then(|| self.buffer()[index] == 1) {
+            Some(b) => Scalar::new(
+                vortex_dtype::DType::Bool(self.validity().nullability()),
+                vortex_scalar::ScalarValue::Bool(b),
+            ),
+            None => Scalar::null(self.dtype().clone()),
+        };
+
+        Ok(scalar)
+    }
+}
+
+impl AsArrowArray for ByteBoolArray {
+    fn as_arrow(&self) -> VortexResult<ArrowArrayRef> {
+        let nulls = self.logical_validity().to_null_buffer()?;
+
+        let mut builder = BooleanBufferBuilder::new(self.len());
+        builder.append_slice(self.as_ref());
+
+        Ok(Arc::new(ArrowBoolArray::new(builder.finish(), nulls)))
+    }
+}
+
+impl SliceFn for ByteBoolArray {
+    fn slice(&self, start: usize, stop: usize) -> VortexResult<crate::Array> {
+        let length = stop - start;
+
+        let validity = self.validity().slice(start, stop)?;
+
+        let slice_metadata = Arc::new(ByteBoolMetadata {
+            validity: validity.to_metadata(length)?,
+            length,
+        });
+
+        ArrayData::try_new(
+            self.encoding(),
+            self.dtype().clone(),
+            slice_metadata,
+            Some(self.buffer().slice(start..stop)),
+            validity
+                .into_array_data()
+                .into_iter()
+                .collect::<Vec<_>>()
+                .into(),
+            StatsSet::new(),
+        )
+        .map(crate::Array::Data)
+    }
+}
+
+impl TakeFn for ByteBoolArray {
+    fn take(&self, indices: &crate::Array) -> VortexResult<crate::Array> {
+        let validity = self.validity();
+        let indices = indices.clone().flatten_primitive()?;
+        let bools = self.as_ref();
+
+        let bools = match_each_integer_ptype!(indices.ptype(), |$I| {
+            indices.typed_data::<$I>()
+            .iter()
+            .map(|&idx| {
+                let idx = idx.as_();
+                if validity.is_valid(idx) {
```

Comment:
It might be better to first match on validity and short-cut for AllValid / AllInvalid.

Then convert validity to a null_buffer, and `iter().zip()`.

Currently this makes a function call for every element, which can be quite expensive.